### ISSUE TITLE
OpenStreetMap login error with F-Droid version

### ISSUE THREAD

**Issue by rugk** [organicmaps/organicmaps#818]
I get the error in https://github.com/organicmaps/organicmaps/issues/451 / https://github.com/organicmaps/organicmaps/issues/451 again, i.e. after entering the credentials a simple popup shows up:
> Login error

## Problem

Likely the credentials you've added in https://github.com/organicmaps/organicmaps/pull/467, which should fix this, are now invalid?

## System

v2021.06..23-9-gc3f05a1646-FDroid
data version: 210529

**Comment by rugk** [comment]
Ah I just see it's likely a duplicate of https://github.com/organicmaps/organicmaps/issues/797

The issue title says iOS, but it also seems to affect Android.

**Comment by biodranik** [comment]
> Likely the credentials you've added in #467, which should fix this, are now invalid?

No, the problem was on the OSM side, this commit fixed it: 00cf8f3ff49abe1302b671632adaf5966fb35daa

Who is responsible for publishing an F-Droid update? Ping him, please :)

**Comment by rugk** [comment]
> Who is responsible for publishing an F-Droid update? Ping him, please :)

Depending on your build process, F-Droid does it automatically.

So based on your commit https://github.com/organicmaps/organicmaps/releases/tag/2021-07-03 should include that fix… Well, then we'll need to wait.

**Comment by rugk** [comment]
Ah, searching further:

> Who is responsible for publishing an F-Droid update? Ping him, please :)

[Here is how to enable automatic updates](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9126#note_594113959), apparently you need to bump those values:

> Bump version name and version code in AndroidManifest.xml before tagging each release

If that does not solve it, ask the people [on F-Droid](https://gitlab.com/fdroid/rfp/-/issues/1665) or in your issue #33.

**Comment by rugk** [comment]
Oh okay, [just commented](https://github.com/organicmaps/organicmaps/issues/33#issuecomment-876349674)…

**Comment by relan** [comment]
> So based on your commit https://github.com/organicmaps/organicmaps/releases/tag/2021-07-03 should include that fix… Well, then we'll need to wait.

2021.07.04-2 is on its way: https://monitor.f-droid.org/builds/log/app.organicmaps/21070416

**Comment by biodranik** [comment]
It's better to publish https://github.com/organicmaps/organicmaps/releases/tag/2021.07.08-3 in F-Droid as it has an updated OSM data.

It would be great if someone from F-Droid could monitor our tags or betas and publish relevant releases :)

**Comment by rugk** [comment]
> It would be great if someone from F-Droid could monitor our tags or betas and publish relevant releases :)

@biodranik F-Droid is unfortunately as community project with way too many resources as you are, so no"one" can do that manually…
However, as I said in https://github.com/organicmaps/organicmaps/issues/33#issuecomment-876349674:
> AFAIK you can [choose another update mode](https://f-droid.org/en/docs/Build_Metadata_Reference/#UpdateCheckMode) based on git tags, e.g. instead, if that better suits.

I.e. you can change the update mode to automatically update based on tag changes instead of source code stuff.

**Comment by biodranik** [comment]
As far as I understand, the F-Droid approach can't use our version which is autogenerated by build.gradle script.

**Comment by rugk** [comment]
Correct. So should I open a new issue about that "How to get F-Droid auto-updates?", which seems to be what this issue now morphed into here…

**Comment by relan** [comment]
> It's better to publish https://github.com/organicmaps/organicmaps/releases/tag/2021.07.08-3 in F-Droid as it has an updated OSM data.

Added 2021.07.08-3: https://gitlab.com/fdroid/fdroiddata/-/commit/46c36bde7a0ac5a1b7e456cf44cbf247497ec735

> It would be great if someone from F-Droid could monitor our tags or betas and publish relevant releases :)

I do this as the time allows (after building and testing each release of course).

> F-Droid is unfortunately as community project

It's _fortunately_ a community project, so _you_ can become a maintainer of Organic Maps metadata on F-Droid ;)

**Comment by rugk** [comment]
Okay, finally opened an issue here: https://github.com/organicmaps/organicmaps/issues/825

**Comment by rtsisyk** [comment]
@rugk @relan thank you for collaboration. OSM auth problem has been fixed, so I'm closing this ticket. Let's track automatic updates for F-Droid in #825.
